language/tools/move-bytecode-utils/src/layout.rs

```diff
@@ -158,55 +174,115 @@ impl<'a, T: GetModule> SerdeLayoutBuilder<'a, T> {
         module_id: &ModuleId,
         name: &Identifier,
         type_arguments: &[Format],
-        shallow: bool,
     ) -> Result<Format, T::Error> {
         // build a human-readable name for the struct type. this should do the same thing as
         // StructTag::display(), but it's not easy to use that code here
-        let struct_key = if type_arguments.is_empty() {
-            format!("{}::{}", module_id, name)
+
+        let declaring_module = self
+            .module_resolver
+            .get_module_by_id(module_id)?
+            .expect("Failed to resolve module");
```

Comment:
I see the function returns a `Result`. Should these panics and asserts be errors?